sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -1014,6 +1018,42 @@ def from_string(cls, permission):
         return parsed
 
 
+class ContainerCpkScopeInfo(ContainerCpkScopeInfoGen):
+    """This scope is then used implicitly for all future writes within the container,
+    but can be overridden per-request via explicit request headers.
+
+    :param default_encryption_scope: Optional. Version 2019-07-07 and later.
+        Specifies the default encryption scope to set on the container and use for
+        all future writes.
+    :type default_encryption_scope: str
+    :param deny_encryption_scope_override: Optional. Version 2019-07-07 and
+        newer.  If true, prevents any request from specifying a different
+        encryption scope than the scope set on the container.
+    :type deny_encryption_scope_override: bool
+    """
+
+    def __init__(self, **kwargs):
+        super(ContainerCpkScopeInfo, self).__init__(**kwargs)
+        self.default_encryption_scope = kwargs.get('default_encryption_scope', None)
+        self.deny_encryption_scope_override = kwargs.get('deny_encryption_scope_override', False)
+
+
+class CpkScopeInfo(CpkScopeInfoGen):
+    """This indicates the encryption scope that should be used to encrypt the application data
+
+    :param encryption_scope: Optional. Version 2019-07-07 and later.
+         Specifies the name of the encryption scope to use to encrypt the data
+         provided in the request. If not specified, encryption is performed with
+         the default account encryption scope.  For more information, see
+         Encryption at Rest for Azure Storage Services.
+    :type encryption_scope: str
+    """
+
+    def __init__(self, **kwargs):
+        super(CpkScopeInfo, self).__init__(**kwargs)
+        self.encryption_scope = kwargs.get('encryption_scope', None)
```

Comment:
@johanste - I would like to remove this model and flatten it out, so that in the operations we just specify:
```
client.create_append_blob(encryption_scope="my-scope")
```
However in this case - `encryption_scope` can have two "values", as specifying the object without explicit scope means "use the default scope for the container"
I was thinking of separating this into two parameters for the two behaviours:
```
client.create_append_blob(use_default_encryption_scope=True)
client.create_append_blob(encryption_scope="override_value"
```
Do we have guidance or precedent on this scenario?
Thoughts?

Reply:
Resolved OOB - opened a separate PR to deal with API changes
https://github.com/Azure/azure-sdk-for-python/pull/9691